### PR TITLE
feat(sdk): adding bucket cors rule

### PR DESCRIPTION
**Pull request by tsuf239**
Allowing this app: https://github.com/winglang/react-file-upload-example to work, as requested by a potential wing user :)

## Checklist

- [x] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [x] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

File: examples/tests/sdk_tests/bucket/cors.test.w [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+bring expect;
+
+
+let api = new cloud.Api();
+
+let bucket = new cloud.Bucket(public: true);
+
+bucket.addCorsConfiguration(
+  allowedMethods: [http.HttpMethod.GET],
+  allowedOrigins:[api.url]
```
Comment (Chriscbr) [inline examples/tests/sdk_tests/bucket/cors.test.w:13]:
```suggestion
  allowedOrigins: [api.url]
```

File: examples/tests/sdk_tests/bucket/cors.test.w [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+);
+
+bucket.addObject("hello", "hello");
+
+
+api.get("/test-bucket-cors", inflight (req) => {
+  let request = util.shell("curl -v --request OPTIONS '{bucket.signedUrl("hello")}' -H 'Origin: {api.url}' -H 'Access-Control-Request-Method: GET'");
+  return {
+    body: request
+  };
```
Comment (Chriscbr) [inline examples/tests/sdk_tests/bucket/cors.test.w:23]: Is it possible to use `http.fetch` instead? Perhaps like

```suggestion
  let response = http.fetch(
    bucket.signedUrl("hello"),
    method: http.HttpMethod.OPTIONS,
    headers: {
      Origin: api.url,
      "Access-Control-Request-Method": "GET",
    },
  );
  return {
    body: response.body
  };
```

File: libs/wingsdk/src/cloud/bucket.ts [hunk 3]
```diff
@@ -46,6 +47,42 @@ export interface BucketProps {
   readonly public?: boolean;
 }
 
+export interface BucketCorsOptions {
```
Comment (Chriscbr) [inline libs/wingsdk/src/cloud/bucket.ts:50]: One of the build errors is asking for a docstring for this interface

File: packages/@winglang/sdk/src/cloud/bucket.ts [hunk 4]
```diff
@@ -46,6 +47,45 @@ export interface BucketProps {
...
+   * @example ["Content-Type"]
+   * @default - []
+   */
+  readonly exposeHeaders?: Array<string>;
+
+  /**
+   * How long the browser should cache preflight request results.
+   * @default - 300 seconds
+   */
+  readonly maxAgeSeconds?: Number;
```
Comment (Chriscbr) [inline packages/@winglang/sdk/src/cloud/bucket.ts:86]:
```suggestion
  /**
   * How long the browser should cache preflight request results.
   * @default - 300 seconds
   */
  readonly maxAge?: Duration;
```

File: packages/@winglang/sdk/src/cloud/bucket.ts
Comment (Chriscbr) [inline packages/@winglang/sdk/src/cloud/bucket.ts:1]: Can we add a `cors: boolean` option to the constructor which defaults to `true`? The `true` setting should be equivalent to adding the following config:

```
{
    allowHeaders: ["*"],
    allowOrigins: ["*"],
    allowMethods: ["DELETE", "GET", "HEAD", "POST", "PUT"],
    exposeHeaders: [],
    maxAge: 0s,
}
```
(these are the same defaults SST uses: https://sst.dev/docs/component/aws/bucket#cors)

**Review by Chriscbr — CHANGES_REQUESTED** [review]

**Comment by tsuf239** [comment]
@Chriscbr thanks for the addition,
is this PR still needed?

**Pull request closed without merging**